src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/Amd64/GcInfo.cs

```diff
@@ -380,9 +380,17 @@ private List<InterruptibleRange> EnumerateInterruptibleRanges(byte[] image, int
         /// GcInfo version is 1 up to ReadyTorun version 1.x.
         /// GcInfo version is current from  ReadyToRun version 2.0
         /// </summary>
-        private int ReadyToRunVersionToGcInfoVersion(int readyToRunMajorVersion)
+        private int ReadyToRunVersionToGcInfoVersion(int readyToRunMajorVersion, int readyToRunMinorVersion)
         {
-            return (readyToRunMajorVersion == 1) ? 1 : GCINFO_VERSION;
+            if (readyToRunMajorVersion == 1)
+                return 1;
+
+            // R2R 9.0 and 9.1 use GCInfo v2
+            // R2R 9.2 uses GCInfo v3
+            if (readyToRunMajorVersion == 9 && readyToRunMinorVersion < 2)
```

Comment:
Since v1 is handled here, should we also handle v2-v8? (same for the gcinfo.h version of the code)

Reply:
There are really just 3 versions of gc info - 1,2,3 (and 3 is the same format as 2, just some extra info, but can be used by old runtimes). 
This maps r2r versions to gc info versions. 

Not sure if this part needs to support all versions, maybe for introspection. The runtime does not support r2r versions below 9.

Reply:
`ILCompiler.Reflection.ReadyToRun` is meant to handle old format versions. 

Also, `GCINFO_VERSION` earlier in this file should be bumped to 3 (or deleted - the constant does not really make sense here).